homework/ba-nguyen/lesson-7/max-area-of-island.ts

```diff
@@ -0,0 +1,31 @@
+function maxAreaOfIsland(grid: number[][]): number {
+  const nr = grid.length;
+  const nc = grid[0].length;
+  let count = 0;
+  let res = 0;
+
+  for (let i = 0; i < nr; i++) {
+    for (let j = 0; j < nc; j++) {
+      count = 0;
+      dfs(i, j);
+      res = Math.max(res, count);
+    }
+  }
+
+  function dfs(r: number, c: number) {
+    if (!grid?.[r]?.[c]) return;
+
+    const dr: number[] = [1, -1, 0, 0];
```

Comment:
should we make it global var instead of redefining in every dfs function?

Reply:
oke i see the problem here, let me update the code